docs/deploying.md

```diff
@@ -1,35 +1,419 @@
-# Deploying
+page_title: Deploying a registry service
+page_description: Explains how to deploy a registry service
+page_keywords: registry, service, images, repository
 
-**TODO(stevvooe):** This should discuss various deployment scenarios for
-production-ready deployments. These may be backed by ready-made docker images
-but this should explain how they were created and what considerations were
-present.
+# Deploying a registry service
 
+This section explains how to deploy a Docker Registry Service either privately
+for your own company or publicly for other users. For example, your company may
+require a private registry to support your continuous integration (CI) system as
+it builds new releases or test servers. Alternatively, your company may have a
+large number of products or services with images you wish to server in a branded
+manner.
 
-# Middleware Configuration
+Docker's public registry maintains a default `registry` image to assist you in the
+deployment process. This registry image is sufficient for running local tests
+but is insufficient for production. For production you should configure and
+build your own custom registry image from the `docker/distribution` code.
 
-This section describes how to configure storage middleware in the registry to enable layers to be served via a CDN, thus reducing requests to the storage layer.  Currently [Amazon Cloudfront](http://aws.amazon.com/cloudfront/) is supported and must be used in conjunction with the S3 storage driver.
 
-## Cloudfront
+## Simple example with the official image
 
-## Parameters
+In this section, you create a local registry using Docker's official image. You
+push an image to, and then pull the same image from, the registry. This a good
+exercise for understanding the basic interactions a client has with a
+local registry.
 
-`name`: The name of the storage middleware.  Currently `cloudfront` is an accepted value.
+1. Install Docker.
 
-`disabled`: This can be set to false to easily disable the middleware.
+2. Run the `hello-world` image from the Docker public registry.
 
-`options` : A set of key/value options to configure the middleware:
+		$ docker run hello-world
+	
+	The `run` command automatically pulls the image from Docker's official images.
 
-* `baseurl` : The cloudfront base URL
-* `privatekey` : The location of your AWS private key on the filesystem 
-* `keypairid` : The ID of your Cloudfront keypair.
-* `duration` : The duration in minutes for which the URL is valid.  Default is 20.
+3. Start a registry service on your localhost.
 
-Note: Cloudfront keys exist separately to other AWS keys.  See [here](http://docs.aws.amazon.com/AWSSecurityCredentials/1.0/AboutAWSCredentials.html#KeyPairs) for more information.
+		$ docker run -p 5000:5000 registry
+		
+	This starts a registry on your `DOCKER_HOST` running on port `5000`. 
+	
+3. List your images.
 
-## Example
+		$ docker images
```

Comment:
This is a little odd to reformat shell output with an html table. It looks like there is a css class but I would just use a "```" block and keep the original shell output formatting:

```shell
$ docker images
REPOSITORY          TAG                       IMAGE ID            CREATED             VIRTUAL SIZE
registry2           latest                    4247f6c17fcd        2 days ago          1.022 GB
<none>              <none>                    ae7d60b9b98c        2 days ago          1.022 GB
<none>              <none>                    dfacdd2afc71        2 days ago          1.022 GB
<none>              <none>                    f4ce2a445ed9        2 days ago          1.022 GB
<none>              <none>                    ed60126df226        2 days ago          1.022 GB
<none>              <none>                    ba8f4a533ce2        2 days ago          1.022 GB
<none>              <none>                    063a5f8dac1a        2 days ago          1.022 GB
<none>              <none>                    b3312f8a689c        2 days ago          1.022 GB
<none>              <none>                    58e33b221404        2 days ago          1.022 GB
docker              address-digest-deadlock   05f7ba10c290        2 weeks ago         1.54 GB
...
```

This is less surprising. Please disregard if these don't follow our documentation guidelines.